cmd/tink-cli/cmd/get/get.go

```diff
@@ -0,0 +1,100 @@
+package get
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/jedib0t/go-pretty/table"
+	"github.com/spf13/cobra"
+)
+
+type CmdOpt struct {
+	// Headers is the list of headers you want to print as part of the list
+	Headers []string
+	// RetrieveData reaches out to Tinkerbell and it gets the required data
+	RetrieveData func(context.Context) ([]interface{}, error)
+	// PopulateTable populates a table with the data retrieved with the RetrieveData function.
+	PopulateTable func([]interface{}, table.Writer) error
+
+	// Output specifies the output you want the list of resources printed
+	// out. By default it is table but it can be JSON ar CSV.
+	Output string
+	// NoHeaders does not print the header line
+	NoHeaders bool
+}
+
+const shortDescr = `
+Display one or many resources`
+
+const longDescr = `
+Prints a table contains the most important information about a specific
+resource. You can specify the kind of output you want to receive. It can be
+table or csv or json.
+`
+
+const exampleDescr = `
+# List all hardware in table output format.
+tink hardware get
+
+# List all workflow in csv output format.
+tink template get -o csv
+
+# List a single template in json output format.
+tink workflow get -o json [id]
+`
+
+func NewGetCommand(opt CmdOpt) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                   "get",
+		Short:                 shortDescr,
+		Long:                  longDescr,
+		Example:               exampleDescr,
+		DisableFlagsInUseLine: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			t := table.NewWriter()
+			t.SetOutputMirror(cmd.OutOrStdout())
+
+			data, err := opt.RetrieveData(cmd.Context())
+			if err != nil {
+				return err
+			}
+
+			if !opt.NoHeaders {
+				header := table.Row{}
+				for _, h := range opt.Headers {
+					header = append(header, h)
+				}
+				t.AppendHeader(header)
+			}
+
+			// TODO(gianarb): Technically this is not needed for
+			// all the output formats but for now that's fine
+			if err := opt.PopulateTable(data, t); err != nil {
+				return err
+			}
+
+			switch opt.Output {
+			case "json":
+				// TODO(gianarb): the table library we use do
+				// not support JSON right now. I am not even
+				// sure I like tables! So complicated...
+				b, err := json.Marshal(struct {
+					Data interface{} `json:"data"`
+				}{Data: data})
```

Comment:
I don't know, I am thinking about a future where we will have pagination?

Reply:
hmm yeah I see that case. I tend to favor jsonnl and forego pagination and leave that to either filtering things or head/tail/grep. But this is fine too.

Reply:
I don't expect pagination added by the cli, I think I would expect the raw API response. The `packet-cli` confuses me in this way because it outputs the `packngo` Go structure of the API responses rather than the raw API responses.

Thinking about `kubectl` again, I think they also prefer the raw API responses (which is part of the reason we get managed-fields noise).

Reply:
I suppose that's not possible when the API does not speak json natively.